Updated Hello_AVH example

Evagan2018 opened the pull request:
Migrated to the new GitHub CI workflow.
Added build steps for Corestone 300, 310, and 315 by using the AC6 build tools.
Added executed steps for generated Corestone 300, 310, and 315 contexts.

Remarks:
(1) For executing the Corestone_315_AC6 on GHB the second display (hdlcd) should be disabled
(2) Capturing simulation data has been not implemented yet. Under investigation.
(3) Simulation Artifacts for post run analysis not defined
(4) GCC toolchain not supported yet.

Commit:
Corrected context for Corestone 315 with AC6

Commit:
Save current work. Improved build and run descriptions.

Commit:
Save current work. Introduce new CI workflow file (initial state - work in progress).

Commit:
Added ARMCM55_config.txt configuration file (under development) for the FVP_Corstone_SSE-300 model.

Commit:
Added Build section which include the build steps for the contexts: Corstone_300, Corstone_310, and Corstone_315 by using the AC6 compiler.
Added Execute section for executing the Corstone_300 context. This is part is under development.

Commit:
Corrected commandlines for building a specific context.

Commit:
Added entries for cmake and ninja
Corrected settings for armclang and avh-fvp

Commit:
Corrected settings for building projects.
Added simulation limit for executing the models.

Commit:
Updated GitHub workflow script.

Commit:
Corrected yml syntax errors

Commit:
Fixed typo.

Commit:
removed old workflow

Commit:
Removed read write permissions. Try to solve a deadsneak failure on GitHub runner.

Commit:
Comment out
       - name: Install system packages
         run: |
             sudo add-apt-repository ppa:deadsnakes/ppa
             sudo apt-get install libpython3.9 libtinfo5

Commit:
Set working directory for build

Commit:
Added working-directory for the step: Execute context Corstone_300_AC6

Commit:
Removed commented out code. Added a diagnosis step only for a short test. This will be removed later.

Commit:
Removed diagnosis step

Commit:
Added steps for Corestone 310 and 315 models.

Commit:
Corrected model strings: FVP_Corestone_SSE-310 and FVP_Corestone_SSE-315

Commit:
Corrected settings for Corestone 315 FVP

Commit:
Updated visualization settings for Corestone 300, 310, and 315

Commit:
Added additional visualization setting for Corestone 315.

Commit:
Added upload of Corstone_300_AC6.avh.log Artifact

Commit:
Changed setting for Artifact

Commit:
Experimental: Install missing xterm on Ubuntu latest

Commit:
Experimental: Added action to install xvfb Virtual Framebuffer on Ubuntu. It is needed for xterm

Commit:
Adding GHB workflow steps for building and executing the hello_vsi example by using the Corestone_300 model and the GCC.

Commit:
Removed previous changes:  Adding GHB workflow steps for building and executing the hello_vsi example by using the Corestone_300 model and the GCC.
GCC is not supported yet as toolchain for this project

vovamarch commented on hello_vsi/README.md:
```diff
@@ -1,32 +1,68 @@
+
+To build the **Corstone-300** target with **AC6** (default toolchain): 
+
+short form
```
No need for short form here. Users need to know cbuild syntax.

vovamarch commented on hello_vsi/README.md:
````diff
@@ -1,32 +1,68 @@
+cbuild hello_vsi.csolution.yml --update-rte --packs --context hello_vsi.event+Corstone_300 --toolchain AC6
+```
+
+To build the **Corstone-315** target with **AC6**:
````
add --toolchain AC6 to be specific about toolchain;

vovamarch commented on hello_vsi/README.md:
````diff
@@ -1,32 +1,68 @@
 ```
 
+
+target **Corstone_315** which has been build with **AC6**  for **60 sec** use:
````
No need to duplicate, it is clear that different model and firmware shall be used for different target.

Commit:
.github\workflows\hello_vsi_ci.yml
    - Corrected cbuild commandline: Added -r for rebuilding. This avoids the usage of build Artifacts from previous builds.
      Build errors have been reported in case of building consecutively the same context with different compilers.
    - Added build and execution section for Corstone_300, Corstone_310, and Corstone_315 contexts by using GCC.

  hello_vsi\cdefault.yml
    - Added section for the GCC

  hello_vsi.cproject.yml
    - Removed entries related to the GUI context

  hello_vsi.csolution.yml
    - Removed entries related to the GUI context

Commit:
Added step for uploading FVP simulation results to GitHub after executing the workflow. These results will be available as Artifacts.

Commit:
vcpkg-configuration.json
 Corrected entry

hello_vsi/cdefault.yml
  Improved structure
  Added warnings for GCC

hello_vsi/hello_vsi.cproject.yml
  Removed ARM 2D entries

hello_vsi/hello_vsi.csolution.yml
   Added output-dirs: section

.github/workflows
   Corrected output directories. Introduced AC6 and GCC subdirectories.

Commit:
Added diagnosis step

Commit:
Create action for tree diagnosis.

Commit:
* Create action for tree diagnosis.

Commit:
Corrected call of tree command

Commit:
Removed diagnosis steps with tree command

Commit:
Added step (experimental): Show directory structure of ./out folder

Commit:
Added erroneously deleted build actions.

Commit:
hello_vsi.csolution.yml : Added ARM::CMSIS@6.1.0, ARM::CMSIS-Compiler@2.1.0, and ARM::RTX@5.9.0 pack to the pack list
hello_vsi.cproject.yml  : Added setup: Corstone_315 GCC

Commit:
Added workflow with matrix build (alpha state)

Commit:
Corrected path setting

Commit:
Reworked mps_board selection

Commit:
Modified workflows

Commit:
Modified amount of parallel jobs for the Matrix build

Commit:
Modified multi-line implementation.

Commit:
Modified multi-line implementation.

Commit:
Corrected initialization for FVP

Commit:
FVP_Corestone_SSE-300.cfg, FVP_Corestone_SSE-310.cfg, and FVP_Corestone_SSE-315.cfg
    Added configuration files for each FVP model.

  hello_vsi_ci_matrix.yml
    Modified workflow file to support configuration files.
    Changed name of simulation output file. It using now the FVP model name.

Commit:
hello_vsi_ci_matrix.yml
   Corrected filename which contains the simulation results.

Commit:
Corrected names of the FVP configuration files (typo).

Commit:
Shorten qualifier names and descriptions
Added step for uploading the build log

Commit:
Added diagnosis step for debugging

Commit:
Corrected typos

Commit:
Redirect build log to file.

Commit:
Added - --entry=Reset_Handler to the Link section for GCC (experimental) for enabling the simulation when the model gets executed.

Commit:
Experimental setup (modifications under test) for stack_base and stack_limit in case of gcc and semihosting.
Modified configuration only for FVP_Corstone_SSE-300.

Commit:
Experimental setup (modifications under test) for stack_base and stack_limit in case of gcc and semihosting.
 Added initialization for the semihosting-heap_base and semihosting-heap_limit.
 Modified configuration only for FVP_Corstone_SSE-300.

Commit:
* Experimental setup (modifications under test) for stack_base and stack_limit in case of gcc and semihosting.
  Modified configuration only for FVP_Corstone_SSE-300.
  Set Secure vector-table offset at reset and disabled ARMv8-M Security Extensions.

Commit:
Improved formatting.

Commit:
hello_vsi_ci.yml: Added configuration files for each FVP.

jkrech commented on vcpkg-configuration.json:
```diff
@@ -2,20 +2,16 @@
-	  "arm:models/arm/avh-fvp": "^11.22.39",
-	  "microsoft:tools/kitware/cmake": "^3.25.2",
-	  "microsoft:tools/ninja-build/ninja": "^1.10.2"
+	  "arm:tools/kitware/cmake": "3.28.4",
```
Prior to this change you have configured a compatible min version using prepending `^`. Now you are requiring a fixed version of all tools, compilers and models. Is that intentional? In that case remove `^` from arm-none-eabi-gcc as well.

Evagan2018 replied:
Has been solved.

jkrech commented on hello_vsi/hello_vsi.csolution.yml:
```diff
@@ -1,13 +1,28 @@
+  description: Hello example for the Video Streaming Interface
+
+  packs:
+    - pack: ARM::CMSIS@6.0.0
```
Why would you specify multiple pack versions of the same pack concurrently. If the components are without version, the component version from the most recent selected pack will be selected. 
Take a look at the `hello_vsi.cbuild-pack.yml` file to see which pack versions are actually used.

Evagan2018 replied:
After a review with Vladimir the pack requirements have been moved to the hello_vsi.cproject.yml file.
Dedicasted component versions have been set for this project.

jkrech commented on hello_vsi/hello_vsi.csolution.yml:
```diff
@@ -1,13 +1,28 @@
+  packs:
+    - pack: ARM::CMSIS@6.0.0
+    - pack: ARM::CMSIS@6.1.0
+    - pack: ARM::CMSIS-Compiler@2.0.0
```
If you take a look in the file `hello_vsi.cproject.yml` you will see that 
- ARM::CMSIS@6.1.0
- ARM::CMSIS-RTX@5.9.0
- ARM::CMSIS-Compiler@2.1.0
are unconditionally selected as well
and 
- ARM::SSE_315_BSP@1.0.0
- ARM::V2M_MPS3_SSE_300_BSP@1.5.0
- ARM::V2M_MSP3_SSE_310_BSP@1.4.0
are selected for the respective target types.
What does that achieve?

Evagan2018 replied:
Has been corrected as follow:

  **packs:**
    - pack: ARM::CMSIS@6.1.0
    - pack: ARM::CMSIS-RTX@5.9.0
    - pack: ARM::CMSIS-Compiler@2.1.0

    - pack: ARM::V2M_MPS3_SSE_300_BSP@1.5.0
      for-context: +Corstone_300
    - pack: ARM::V2M_MPS3_SSE_310_BSP@1.4.0
      for-context: +Corstone_310
    - pack: ARM::SSE_315_BSP@1.0.0
      for-context: +Corstone_315
    - pack: ARM::DMA350@1.0.0
      for-context: +Corstone_315

jkrech commented on hello_vsi/README.md:
```diff
@@ -1,32 +1,68 @@
+Use the cbuild tool or an IDE to build the application project in csolution format (see the project [README](../README.md)).
 
-All applications have been ported for Corstone-300, Corstone-310 and Corstone-315, and can be compiled with Arm Compiler 6 or GCC. By default the Arm Compiler 6 is used, to compile with GCC, use `--toolchain GCC` option for cbuild command.
+All applications have been ported for **Corstone-300**, **Corstone-310** and **Corstone-315**, and can be compiled with **Arm Compiler** 6 or **GCC**. By default the Arm Compiler 6 is used, to compile with GCC, use `--toolchain GCC` option for the cbuild command.
```
To make this "symmetric" you could choose not to specify a `compiler:` node in the `csolution.yml` file and use `--toolchain AC6` and `--toolchain GCC` at the command line instead?

Evagan2018 replied:
Has been corrected as follow:

All applications have been ported for **Corstone-300**, **Corstone-310** and **Corstone-315**, and can be compiled with **Arm Compiler 6** or **GCC**. To compile with **AC6** use `--toolchain AC6` and for **GCC** the `--toolchain GCC` option with the cbuild command.

jkrech commented on hello_vsi/FVP_Corstone_SSE-300.cfg:
```diff
@@ -0,0 +1,17 @@
+# Notes:
+# 1) Disable visualization for CI environments.
+# 2) Disable gatting logic for memory
```
gating ?

Evagan2018 replied:
Comment nr.2: Disable gatting ....  has been deleted.

jkrech commented on hello_vsi/FVP_Corstone_SSE-300.cfg:
```diff
@@ -0,0 +1,17 @@
+#---------------------------------------------------------------------------------------------------
+mps3_board.visualisation.disable-visualisation=1      # (bool  , init-time) default = '0'          : Enable/disable visualisation
+mps3_board.DISABLE_GATING=1                           # (bool  , init-time) default = '0'          : Disable Memory gating logic
+cpu0.semihosting-stack_base=0x0                       # (int   , init-time) default = '0x20800000' : Virtual address of base of descending stack
```
I am surprised that the different platforms require different parameters to be included.

Evagan2018 replied:
Aligned to minimum required differences.
i.e. FVP_Corstone_SSE-315 is using mps4_board instead of  mps3_board and requires also to disable the HLCD option.

Commit:
vcpkg-configuration.json
 - Configured the minimum version for all tools, compilers and models.

cdefault.yml
 - Set as default compiler AC6
 - Removed the following linker option for GCC:
   - --specs=rdimon.specs
   Do not use this option with FVP and redirection of the output streams.
   Further info: https://community.arm.com/support-forums/f/compilers-and-libraries-forum/44223/what-are-the-compile-option-differences-between---specs-nosys-specs-and---specs-rdimon-specs-in-arm-none-eabi-gcc

hello_vsi/hello_vsi.csolution.yml
 - Corrected CMSIS-Toolbox version.

Commit:
FVP_Corstone_SSE-300.cfg: Corrected settings.

Commit:
hello_vsi_ci_matrix.yml: Experimental:Get list of contexts

Commit:
hello_vsi_ci_matrix.yml: Experimental:Get list of contexts

Commit:
hello_vsi_ci_matrix.yml: Experimental:Get list of contexts

Commit:
Experimental:Get list of contexts

Commit:
Experimental:Get list of contexts

Commit:
Experimental:Get list of contexts

Commit:
Experimental:Get list of contexts

Commit:
Experimental:Get list of contexts

Commit:
Experimental:Get list of contexts

Commit:
Experimental:Get list of contexts

Commit:
Experimental:Get list of contexts

Commit:
Experimental:Get list of contexts

Commit:
Experimental:Get list of contexts

Commit:
Experimental:Get list of contexts

Commit:
Experimental:Get list of contexts

Commit:
Experimental:Get list of contexts

Commit:
Experimental: split oneline command to multiline

Commit:
Experimental: split one-line command to multi-line. Corrected forward-slash to back-slash

Commit:
Experimental: split one-line command to multi-line. Done also for building the executable.

Commit:
Experimental: Added os-matrix

Commit:
Experimental: Added conditional installation of xterm

Commit:
Experimental: Changed conditional code execution.

Commit:
Experimental: Modified capturing list for list of contexts.

Commit:
Extended conditional

Commit:
Corrected line-continuation

Commit:
Corrected line-continuation

Commit:
Experimental: Launch 2 runners

Commit:
Experimental: Run on two runners.

Commit:
Added the build type as matrix parameter.

Commit:
Corrected issue with model name (error has been introduced with the previous commit)
Removed experimental sections in workflow file

Commit:
Corrected file names for build and simulation log.

Commit:
Removed actions:
  Install xterm on Ubuntu,
  Install xvfb Virtual Framebuffer on Ubuntu # Needed for xterm
  Build context windows ${{ matrix.btype }}_${{ matrix.target.type }}_${{ matrix.compiler.name }}
  Execute context windows ${{ matrix.btype }}_${{ matrix.target.type }}_${{ matrix.compiler.name }} on ${{ matrix.target.model }}

Reworked section:
  CI_test_run:
    strategy:

  - Removed windows-latest from OS
  - Restructured matrix fields

Commit:
Corrected issue:
   Error when evaluating 'runs-on' for job 'CI_test_run'. .github/workflows/hello_vsi_ci_matrix.yml (Line: 39, Col: 14): Unexpected value 'ubuntu-latest'

Commit:
Simplified:
  - This workflow runs only on ubuntu-latest.
  - Removed the matrix variable ubuntu.
    support any other OS.
  - Removed any occurrence of ubuntu in strings.

Commit:
Removed:
  - from the current workflow matrix the device entries which are never used
  - continue-on-error: true
Added:
  - project path to the workflow trigger section
  - ${{env.PROJECT}} variable

Commit:
Added workflow script to the trigger section

Commit:
Reverted previous changes

Commit:
Removed environment variable.

Evagan2018 committed:
Update main.c

Test

Commit:
Removed hello_vsi_ci.yml workflow file.

Commit:
Renamed workflow file.

Commit:
hello_vsi.csolution.yml  : Corrected pack list
vcpkg-configuration.json : Modified formatting

Commit:
Corrected pack sections.

Commit:
Corrected pack sections.

Commit:
Corrected pack sections.

Commit:
Removed ARMCM55_config.txt not needed for the project.

vovamarch commented on hello_vsi/FVP_Corstone_SSE-300.cfg:
```diff
@@ -0,0 +1,10 @@
+# instance.parameter=value                            #(type, mode)         default = 'def value'  : description : [min..max]
+#---------------------------------------------------------------------------------------------------
+mps3_board.visualisation.disable-visualisation=1      # (bool  , init-time) default = '0'          : Enable/disable visualisation
+mps3_board.DISABLE_GATING=1                           # (bool  , init-time) default = '0'          : Disable Memory gating logic
```
mps3_board.DISABLE_GATING=1  can be removed, as default setting shall work as well

Evagan2018 replied:
Removed

vovamarch commented on hello_vsi/FVP_Corstone_SSE-310.cfg:
```diff
@@ -0,0 +1,6 @@
+# instance.parameter=value       #(type, mode) default = 'def value' : description : [min..max]
+#---------------------------------------------------------------------------------------------------
+mps3_board.visualisation.disable-visualisation=1      # (bool  , init-time) default = '0'          : Enable/disable visualisation
+mps3_board.DISABLE_GATING=1                           # (bool  , init-time) default = '0'          : Disable Memory gating logic
```
mps3_board.DISABLE_GATING=1  can be removed, as default setting shall work as well

Evagan2018 replied:
Removed

vovamarch commented on hello_vsi/FVP_Corstone_SSE-315.cfg:
```diff
@@ -0,0 +1,7 @@
+#---------------------------------------------------------------------------------------------------
+mps4_board.visualisation.disable-visualisation=1      # (bool  , init-time) default = '0'          : Enable/disable visualisation
+vis_hdlcd.disable_visualisation=0                     # (bool  , init-time) default = '0'          : Enable/disable visualisation
+mps4_board.DISABLE_GATING=1                           # (bool  , init-time) default = '0'          : Disable Memory gating logic
```
mps3_board.DISABLE_GATING=1  can be removed, as default setting shall work as well

Evagan2018 replied:
Removed

Commit:
Removed Gatting setting and comments inside the configuration files.

Commit:
Corrected readme.md

Commit:
Corrected cproject.yml and csolution.yml

vovamarch reviewed (approved):
Looks good for merge
